src/nl/hannahsten/texifyidea/completion/LatexFileProvider.kt

```diff
@@ -54,14 +57,22 @@ class LatexFileProvider : CompletionProvider<CompletionParameters>() {
                 .toSet()
                 .forEach { addByDirectory(it, autocompleteText, result) }
 
-        // Add all included graphicspaths.
-        val graphicsPaths = parameters.originalFile.commandsInFileSet().filter { it.commandToken.text == "\\graphicspath" }
-        graphicsPaths.asSequence()
-                .mapNotNull { it.requiredParameter(0) }
-                .mapNotNull { it.println(); baseDirectory.findFileByRelativePath(it) }
-                .filter { it.isDirectory && it != baseDirectory }
-                .toSet()
-                .forEach { addByDirectory(it, autocompleteText, result) }
+        // Add last included graphicspaths.
+        val graphicsPath = parameters.originalFile.commandsInFileSet().last { it.commandToken.text == "\\graphicspath" }
+        graphicsPath.parameterList.first { it1 -> it1.requiredParam != null }
+                .childrenOfType(LatexNormalText::class).forEach { it2 ->
```

Comment:
Instead of using it1 and it2, it's better to give a meaningful name like you would for other variables

Reply:
fine.

Reply:
Should unimplemented iterator names be implemented with a proper name or stay default `it`?

Reply:
When it is clear what it is, that is to say, what `it` is, and you have just one `it`, then you can leave it. So in this case, you could rename `it1` to be a real variable name, and replace `it2 ->` by just using `it`.